Marlin/ultralcd.cpp

```diff
@@ -1969,13 +1969,17 @@ void kill_screen(const char* lcd_msg) {
         print_job_counter.loadStats();
         printStatistics stats = print_job_counter.getStats();
 
-        char timeString[8];
-        sprintf_P(timeString, PSTR("%i:%02i"), int(stats.printTime / 60 / 60), int(stats.printTime / 60) % 60);
+        int min = long(stats.printTime / 60) % 60;
+        int hrs = long(stats.printTime / 60 / 60) % 24;
+        int day = int(stats.printTime / 60 / 60 / 24);
+        char timeString[14];
+        sprintf_P(timeString, PSTR("%i%s %i%s %02i%s"), day, MSG_SHORT_DAY, hrs, MSG_SHORT_HOUR, min, MSG_SHORT_MINUTE);
```

Comment:
This will produce smaller code as it uses less string storage and encourages the compiler to skip creating temporary variables:
```cpp
sprintf_P(timeString, PSTR("%i" MSG_SHORT_DAY " %i" MSG_SHORT_HOUR " %i" MSG_SHORT_MINUTE),
  int(stats.printTime / 60 / 60 / 24),
  int((stats.printTime / 60 / 60) % 24),
  int((stats.printTime / 60) % 60)
);
```